### PR TITLE
Convert Windows to use vckpg instead of nuget

### PR DESCRIPTION
**Pull request by hughbe**
This is better as we don't need to maintain our own dependency on hexchat's GTK and we can rely on the vcpkg community

@qmfrederik FYI

### CODE REVIEW

**Comment by hughbe** [comment]
@filipnavara this should fix the WIn32 issue you mentioned in https://github.com/mono/libgdiplus/pull/269

**Comment by filipnavara** [comment]
There was already Pango in the HexChat package, so it was not a deal-breaker, but I definitely prefer the VCPKG approach.

File: .jenkins-windows.sh [hunk 1]
```diff
@@ -2,16 +2,16 @@
 
 export PATH=/usr/bin:/usr/local/bin:$PATH
 
-git clone https://github.com/Microsoft/vcpkg.git
-bootstrap-vcpkg.bat
-vcpkg integrate install
+git clone https://github.com/Microsoft/vcpkg.git /cygdrive/c/Program\ Files\ \(x86\)/
+/cygdrive/c/Program\ Files\ \(x86\)/bootstrap-vcpkg.bat
+/cygdrive/c/Program\ Files\ \(x86\)/vcpkg integrate install
```
Comment (filipnavara) [inline .jenkins-windows.sh]: This is not writable directory, so it's not going to work even if you get all the path quoting right.

Comment (filipnavara) [reply .jenkins-windows.sh]: You probably want something more like

```
git clone https://github.com/Microsoft/vcpkg.git
vcpkg/bootstrap-vcpkg.bat
vcpkg/vcpkg install giflib libjpeg-turbo libpng cairo glib tiff libexif glib pango
```

**Comment by hughbe** [comment]
@akoeplinger does this require vcpkg to be installed on the mono CI? I'm not sure whether I can do it from the script:

> ./.jenkins-windows.sh: line 6: vcpkg/bootstrap-vcpkg.bat: Permission denied

**Comment by qmfrederik** [comment]
@hughbe Yes, vcpkg is the way to go! The build scripts look a lot nicer now 😄 .

File: src/libgdiplus.vcxproj [hunk 2]
```diff
@@ -87,28 +87,17 @@
   <PropertyGroup Label="UserMacros" />
   <ItemDefinitionGroup>
     <ClCompile>
-      <AdditionalIncludeDirectories>$(ProjectDir)\..;$(ProjectDir)\..\gtk\$(Platform)\release\include;$(ProjectDir)\..\gtk\$(Platform)\release\include\glib-2.0;$(ProjectDir)\..\gtk\$(Platform)\release\lib\glib-2.0\include;$(ProjectDir)\..\gtk\$(Platform)\release\include\cairo;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PreprocessorDefinitions>HAVE_LIBGIF;HAVE_LIBJPEG;HAVE_LIBTIFF;HAVE_LIBPNG;HAVE_FCFINI;_WINDLL;WIN32;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <PreprocessorDefinitions>HAVE_LIBGIF;HAVE_LIBJPEG;HAVE_LIBTIFF;HAVE_LIBPNG;_WINDLL;WIN32;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
Comment (filipnavara) [inline src/libgdiplus.vcxproj]: Where did HAVE_FCFINI go, btw?

**Comment by ras0219-msft** [comment]
We'd also be happy to accept a PR in vcpkg for `libgdiplus` so we can make sure dependency updates don't break you (or at a minimum notify you) :)

**Comment by akoeplinger** [comment]
@hughbe hm, I don't know why it fails that way. If I run the command manually as the same user via RDP it works. I'll need to dig in why :)

**Comment by filipnavara** [comment]
@monojenkins build

**Comment by filipnavara** [comment]
I am not surprised it fails. The "vcpkg integrate install" command install system-wide MSBuild scripts.

**Comment by hughbe** [comment]
Maybe we need to call it just once on the machine (not from the script) to install it?

**Comment by akoeplinger** [comment]
@filipnavara it doesn't fail at `vcpkg integrate install`, it fails right away for some reason.

@hughbe we could do that but since the Windows builders are dynamically provisioned Azure VMs we'd need to do it at provision time. And since vcpkg compiles all the packages that takes quite some time so we'll probably need to use the caching feature instead.

**Comment by akoeplinger** [comment]
Ok, the issue was some weird interaction with the cygwin bash shell calling the Windows batch command.

Moving everything to powershell seemed to work. Build without caching (i.e. compiling all the libs) takes about 30mins now which is OK for now I think, the Travis build takes longer anyway.

Thanks!

**Pull request merged**